Composer/packages/adaptive-form/src/components/fields/IntellisenseFields.tsx

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { FieldProps } from '@bfc/extension-client';
+import { Intellisense } from '@bfc/intellisense';
+import React from 'react';
+
+import { getIntellisenseUrl } from '../../utils/getIntellisenseUrl';
+import { ExpressionSwitchWindow } from '../ExpressionSwitchWindow';
+
+import { JsonField } from './JsonField';
+import { NumberField } from './NumberField';
+import { StringField } from './StringField';
+
+export const IntellisenseTextField: React.FC<FieldProps<string>> = function IntellisenseTextField(props) {
+  const { id, value = '', onChange, uiOptions, focused: defaultFocused } = props;
+
+  const completionListOverrideResolver = (value: string) => {
+    if (value === '') {
+      return <ExpressionSwitchWindow type={'String'} onSwitchToExpression={() => onChange('=')} />;
+    } else {
+      return null;
+    }
+  };
+
+  return (
+    <Intellisense
+      completionListOverrideResolver={completionListOverrideResolver}
+      focused={defaultFocused}
+      id={`intellisense-${id}`}
+      scopes={uiOptions.intellisenseScopes || []}
+      url={getIntellisenseUrl()}
```

Comment:
You want to do this every render?

Reply:
@GeoffCoxMSFT, I believe React is smart enough and do not rerender unless those values changes. Do you see a way to optimize this?